examples/js/postprocessing/EffectComposer.js

```diff
@@ -73,51 +106,54 @@ Object.assign( THREE.EffectComposer.prototype, {
 
 	},
 
-	render: function ( delta ) {
+	render: function ( delta, starti ) {
 
-		var maskActive = false;
+		var maskActive = this.maskActive;
 
 		var pass, i, il = this.passes.length;
 
-		for ( i = 0; i < il; i ++ ) {
+		var scope = this;
 
-			pass = this.passes[ i ];
+		var currentOnAfterRender;
 
-			if ( pass.enabled === false ) continue;
+		for ( i = starti || 0; i < il; i ++ ) {
 
-			pass.render( this.renderer, this.writeBuffer, this.readBuffer, delta, maskActive );
+			pass = this.passes[ i ];
 
-			if ( pass.needsSwap ) {
+			if ( pass.enabled === false ) continue;
 
-				if ( maskActive ) {
+			// If VR mode is enabled and rendering the whole scene is required.
+			// The pass renders the scene and and postprocessing is resumed before
+			// submitting the frame to the headset by using the onAfterRender callback.
+			if ( this.renderer.vr.enabled && pass.scene ) {
 
-					var context = this.renderer.context;
+				currentOnAfterRender = pass.scene.onAfterRender;
 
-					context.stencilFunc( context.NOTEQUAL, 1, 0xffffffff );
+				pass.scene.onAfterRender = function () {
```

Comment:
maybe not create this function here?

Reply:
yeah, good point. The function will be allocated once per frame. I'll factor it out.

Reply:
As is it looks like it would be instantiated `this.passes.length` times? I thought about it but it looked kinda complicated to carry the `i` over.